ssdp.md

```diff
@@ -0,0 +1,135 @@
+
+* New display with presentation receiver service that appears on the network advertises the following SSDP alive message. The new header `FRIENDLY-NAME.example.org` is used for friendly name and `PRESENTATION-ENDPOINT.example.org` for endpoint of the receiver service. The value `urn:example-org:service:osp:1` of header `NT` is used for presentation receiver service.
+
+    ```
+    NOTIFY * HTTP/1.1
+    HOST: 239.255.255.250:1900
+    CACHE-CONTROL: max-age = seconds until advertisement expires
+    LOCATION: URL of the device description
+    NTS: ssdp:alive
+    SERVER: OS/version UPnP/1.0 product/version
+    USN: advertisement UUID
+    NT: urn:example-org:service:osp:1
+    FRIENDLY-NAME.example.org: My Presentation Display
+    PRESENTATION-ENDPOINT.example.org: 192.168.1.100:3000
+    ```
+
+* A controller sends the following SSDP search message to the multicast address. The new header `PRESENTATION-URLS.example.org` allows the controller to send the presentation URLs to the display. 
+
+    ```
+    M-SEARCH * HTTP/1.1
+    HOST: 239.255.255.250:1900
+    MAN: "ssdp:discover"
+    MX: seconds to delay response
+    ST: urn:example-org:service:osp:1
+    PRESENTATION-URLS.example.org: http://example.com/foo.html, http://example.com/bar.html
+    ```
+* A display that can open one of the URLs replies (unicast) with the following SSDP message. The new SSDP header `SUPPORTED-URLS.example.org` contains the URLs the receiver can open from the list of the URLs `PRESENTATION-URLS.example.org` sent in the search SSDP message.
+
+    ```
+    HTTP/1.1 200 OK
+    CACHE-CONTROL: max-age = seconds until advertisement expires
+    DATE: when response was generated
+    LOCATION: URL of the device description
+    SERVER: OS/version UPnP/1.0 product/version
+    USN: advertisement UUID
+    ST: urn:example-org:service:osp:1
+    FRIENDLY-NAME.example.org: My Presentation Display
+    PRESENTATION-ENDPOINT.example.org: 192.168.1.100:3000
+    SUPPORTED-URLS.example.org: http://example.com/foo.html
+    ```
+* The display sends the following SSDP message when the receiver service in not more available. There are no new headers used in the byebye message.
+
+    ```
+    NOTIFY * HTTP/1.1
+    HOST: 239.255.255.250:1900
+    NT: urn:example-org:service:osp:1
+    NTS: ssdp:byebye
+    USN: uuid:advertisement UUID
+    ```
+
+### Remote Playback API: Monitor Display Availability
+TODO (it should be very similar to the Presentation API display availability but for capabilities other parameters related media formats, codecs, encryptions need to be considered)
+
+## Non-functional Requirements
+
+### Reliability
+
+UPnP recommend due to the unreliable nature of UDP to send the SSDP messages more than once (and max three times to avoid network congestion) with some delay of few hundred milliseconds. In addition, the device must re-send its advertisements periodically prior to expiration of the duration specified in
```

Comment:
Chrome desktop sends three search requests spaced 2 seconds apart every 120 seconds.  We've found this is good enough from a reliability perspective, and hope to share some more concrete data.